sdk/logical/error.go

```diff
@@ -61,6 +64,45 @@ var (
 	ErrPathFunctionalityRemoved = errors.New("functionality on this path has been removed")
 )
 
+type DelegatedAuthErrorHandler func(ctx context.Context, initiatingRequest, authRequest *Request, authResponse *Response, err error) (*Response, error)
+
+// Special error indicating the backend wants to delegate authentication elsewhere
+type RequestDelegatedAuth struct {
```

Comment:
Since this is an error type, maybe makes sense to append `Error` to its name?

Also, mind adding 
```
var _ error = &RequestedDelegatedAuth{}
```
?  I like making explicit what interface a type is meant to satisfy.

Reply:
Agreed, implemented  within https://github.com/hashicorp/vault/pull/23924/commits/05012ce747cf33c8fe69ca1cb587443626d9c66a